wagtail/users/templatetags/wagtailusers_tags.py

```diff
@@ -2,13 +2,54 @@
 from collections import defaultdict
 
 from django import template
+from django.contrib.auth import get_permission_codename
+from django.contrib.auth.models import Permission
+from django.utils.text import camel_case_to_spaces
 
 from wagtail import hooks
 from wagtail.users.permission_order import CONTENT_TYPE_ORDER
 
 register = template.Library()
 
 
+def normalize_permission_label(permission: Permission):
+    """
+    Strip model name from the end of the label, e.g. "Can deliver pizza" for a
+    Pizza model becomes "Can deliver". For permissions in the model's
+    Meta.default_permissions with default labels, also replace underscores
+    with spaces.
+
+    This is used to display custom model permissions in the admin.
+
+    See https://github.com/wagtail/wagtail/issues/10982.
+    """
+    label = permission.name
+    content_type = permission.content_type
+    model = content_type.model_class()
+    verbose_name = default_verbose_name = content_type.name
+
+    if model:
+        default_verbose_name = camel_case_to_spaces(model._meta.object_name)
+
+        # If it's in default_permissions and the label matches Django's default
+        # label, remove the model name from the end of the label and replace
+        # underscores with spaces.
+        for action in model._meta.default_permissions:
+            default_codename = get_permission_codename(action, model._meta)
+            is_default = permission.codename == default_codename
+            if is_default and permission.name.startswith(f"Can {action}"):
+                return f"Can {action.replace('_', ' ')}"
```

Comment:
This doesn't seem right - if `action` is the underscored version (and I think it is), then surely the `permission.name.startswith(f"Can {action}")` check isn't going to succeed, as that's the human-friendly version, right? (Unless of course `action` is a single word, in which case replacing underscores is redundant.)

It seems to me this should be something like:
```suggestion
            permission_label_prefix = f"Can {action.replace('_', ' ')}"
            if is_default and permission.name.startswith(permission_label_prefix):
                return permission_label_prefix
```

Reply:
`action` can be anything, but yeah it tends to be underscored version as Django will generate the `Permission` with the codename `{action}_{model_name}`. I tested and Django doesn't seem to care what `action` looks like, e.g. it can even have spaces.

> then surely the `permission.name.startswith(f"Can {action}")` check isn't going to succeed, as that's the human-friendly version, right?

This is where Django falls short, it just takes the `action` as-is and stick that into the `Permission.name` as `Can {action} {verbose_name}`:

https://github.com/django/django/blob/f835787f714cf000603516188b5516f87aa348b1/django/contrib/auth/management/__init__.py#L22-L35

<img width="167" alt="image" src="https://github.com/wagtail/wagtail/assets/6379424/e55974ed-2f32-4e39-bc68-653d8f6b266d">

Unfortunately Django has many issues around permission labels/codenames, hence we're just aiming for a best-effort approach:

- https://code.djangoproject.com/ticket/1688
- https://code.djangoproject.com/ticket/26756
- https://code.djangoproject.com/ticket/27489

I've added b73e34f to demonstrate this.

Reply:
Oops, sorry for the multiple force pushes, I thought the test for `Cause chaos for advanced permission model` should use `ADVANCED` instead of `advanced`, but I just realised that it's a custom label and the label uses the lowercase `advanced` so it was already correct.